packages/kbn-handlebars/index.ts

```diff
@@ -623,70 +646,79 @@ class ElasticHandlebarsVisitor extends Handlebars.Visitor {
     }
 
     return value;
   }
 
-  private setupHelper(node: ProcessableNodeWithPathParts, helperName: string) {
+  private setupHelper(
+    node: ProcessableNode,
+    helperName: string
+  ): {
+    fn?: Handlebars.HelperDelegate;
+    context: any[];
+    params: any[];
+    options: AmbiguousHelperOptions;
+  } {
     return {
       fn: this.container.lookupProperty(this.container.helpers, helperName),
       context: this.scopes[0],
-      params: [...this.resolveNodes(node.params), this.setupParams(node, helperName)],
+      params: this.resolveNodes(node.params),
+      options: this.setupParams(node, helperName),
     };
   }
 
-  private setupDecoratorOptions(
-    decorator: hbs.AST.Decorator | hbs.AST.DecoratorBlock
-  ): Handlebars.HelperOptions {
+  private setupDecoratorOptions(decorator: hbs.AST.Decorator | hbs.AST.DecoratorBlock) {
     // TypeScript: The types indicate that `decorator.path` technically can be an `hbs.AST.Literal`. However, the upstream codebase always treats it as an `hbs.AST.PathExpression`, so we do too.
     const name = (decorator.path as hbs.AST.PathExpression).original;
-    const options = this.setupParams(decorator as hbs.AST.DecoratorBlock, name);
+    const options = toDecoratorOptions(this.setupParams(decorator, name));
 
     if (decorator.params.length > 0) {
       if (!this.processedRootDecorators) {
         // When processing the root decorators, temporarily remove the root context so it's not accessible to the decorator
         const context = this.scopes.shift();
-        // @ts-expect-error: Property 'args' does not exist on type 'HelperOptions'. The 'args' property is expected in decorators
         options.args = this.resolveNodes(decorator.params);
         this.scopes.unshift(context);
       } else {
-        // @ts-expect-error: Property 'args' does not exist on type 'HelperOptions'. The 'args' property is expected in decorators
         options.args = this.resolveNodes(decorator.params);
       }
     }
 
-    // @ts-expect-error: Property 'lookupProperty' does not exist on type 'HelperOptions'
-    delete options.lookupProperty; // There's really no tests/documentation on this, but to match the upstream codebase we'll remove `lookupProperty` from the decorator context
-
     return options;
   }
 
-  private setupParams(
-    node: ProcessableNodeWithPathParts,
-    helperName: string
-  ): Handlebars.HelperOptions {
-    const options: Handlebars.HelperOptions = {
-      // @ts-expect-error: Name should be on there, but the offical types doesn't know this
-      name: helperName,
+  private setupParams(node: ProcessableBlockStatementNode, name: string): Handlebars.HelperOptions;
+  private setupParams(node: ProcessableStatementNode, name: string): NonBlockHelperOptions;
+  private setupParams(node: ProcessableNode, name: string): AmbiguousHelperOptions;
+  private setupParams(node: ProcessableNode, name: string): AmbiguousHelperOptions {
```

Comment:
It's because `ProcessableNode` is a superset of both `ProcessableBlockStatementNode` and `ProcessableStatementNode`. So as far as I can see it should be ok